tests/unit/migrations/flask.sql

```diff
@@ -287,7 +287,7 @@ ALTER SEQUENCE nic_id_seq OWNED BY nic.id;
 
 CREATE TABLE node (
     id integer NOT NULL,
-    label character varying NOT NULL,
+    label character varying NOT NULL UNIQUE,
```

Comment:
Naved and Jeremy suggested that this is not the place that does the migration tests. And the question now is how we can increase the test coverage? Or, where do we do the migration tests?

Reply:
We shouldn't **ever** be modifying this file; the whole point of the tests is to verify that we can cleanly upgrade from an old database schema; this is a snapshot of a database we should be able to upgrade from. What you've done here is manually modify the old schema.

The actual tests are in `hil/migrations.py`.

The missing coverage in `upgrade` is incorrect; we're definitely running that in the migration tests (and it seems to be erroneously missing from all of the other scripts as well). downgrade() is really untested, and we basically haven't done testing with downgrading databases at all.

My inclination is to:

1. Figure out why the upgrade function doesn't look like it's covered
2. Open a separate issue about the fact that we're not testing downgrade() anywhere. Honestly, I'm not sure it's worth the effort to properly support database downgrades, and we've only been adding the function at all because alembic expects it. But we should discuss the issue.

I'm fine opening separate issues for both of these and just overriding the coveralls warning for now.